### PR TITLE
[Checkout] Preventing to pay for cancelled order

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/orderDetails.html.twig [hunk 1]
```diff
@@ -4,7 +4,7 @@
     {% set lastNewPayment = order.getLastNewPayment() %}
     {% include '@SyliusShop/Common/Order/_summary.html.twig' %}
 
-    {% if order.paymentState is not same as ('paid') %}
+    {% if order.paymentState is not same as ('paid') and order.paymentState is not same as ('cancelled') %}
```
Comment (pamil) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/orderDetails.html.twig]: `{% if order.paymentState not in ['paid', 'cancelled'] %}`

Comment (pamil) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/orderDetails.html.twig]: Side note: we should have issers for paid / cancelled / etc. states. The current implementation depends heavily on the state machine, but not on the model. Then we can do `{% if not order.paid and not order.cancelled %}`.

File: src/Sylius/Behat/Resources/config/suites/ui/checkout/paying_for_order.yml [hunk 2]
```diff
@@ -8,19 +8,23 @@ default:
...
                 - sylius.behat.context.setup.channel
                 - sylius.behat.context.setup.geographical
                 - sylius.behat.context.setup.currency
+                - sylius.behat.context.setup.order
                 - sylius.behat.context.setup.payment
                 - sylius.behat.context.setup.shop_security
                 - sylius.behat.context.setup.user
                 - sylius.behat.context.setup.product
                 - sylius.behat.context.setup.shipping
+                - sylius.behat.context.transform.shipping_method
```
Comment (Zales0123) [inline src/Sylius/Behat/Resources/config/suites/ui/checkout/paying_for_order.yml]: Why it's not placed together with other transform contexts? :(

File: features/checkout/paying_for_order/preventing_to_pay_for_cancelled_order.feature [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        And the store allows paying offline
+        And there is a customer "sylius@example.com" that placed an order "#00000022"
+
+    @ui
+    Scenario: Not being able to pay for cancelled order
+        Given the customer bought 3 "Iron Maiden T-Shirt" products
+        And the customer chose "Free" shipping method to "United States" with "offline" payment
+        And this order was cancelled
+        When I want to browse order details for this order
+        Then I should not be able to pay
```
Comment (michalmarcinkowski) [inline features/checkout/paying_for_order/preventing_to_pay_for_cancelled_order.feature]: The customer is not able to pay for the order when `offline` gateway was chosen regardless of the order state. In order to validate this scenario we should choose other than `offline` gateway.

Comment (Arminek) [inline features/checkout/paying_for_order/preventing_to_pay_for_cancelled_order.feature]: If i choose offline gateway i am still able to change the payment method and try to pay again.

Comment (michalmarcinkowski) [inline features/checkout/paying_for_order/preventing_to_pay_for_cancelled_order.feature]: You should be able to change a payment method, but the `pay` button should be hidden for `offline` gateway. E.g. I select "cash on delivery" payment method and then I press `pay` button? Looks strange, what happens then?

**Pull request merged**